packages/fluentui/react-northstar/src/components/Button/Button.tsx

```diff
@@ -100,6 +103,268 @@ export type ButtonStylesProps = Pick<
   hasContent?: boolean;
 };
 
+type GetBorderFocusStylesOptions = {
+  padding: string;
+  radius: string;
+  width: string;
+
+  innerColor: string;
+  outerColor: string;
+
+  zIndex: string;
+};
+
+function getBorderFocusStyles(options: GetBorderFocusStylesOptions) {
+  return {
+    borderColor: 'transparent',
+
+    ':before': {
+      content: '""',
+      position: 'absolute',
+      borderStyle: 'solid',
+      pointerEvents: 'none',
+
+      borderColor: options.innerColor,
+      borderRadius: options.radius,
+      borderWidth: options.width,
+
+      zIndex: options.zIndex,
+
+      top: `calc(0px - ${options.padding})`,
+      bottom: `calc(0px - ${options.padding})`,
+      left: `calc(0px - ${options.padding})`,
+      right: `calc(0px - ${options.padding})`,
+    },
+
+    ':after': {
+      content: '""',
+      position: 'absolute',
+      borderStyle: 'solid',
+      pointerEvents: 'none',
+
+      borderColor: options.outerColor,
+      borderRadius: options.radius,
+      borderWidth: options.width,
+
+      zIndex: options.zIndex,
+
+      top: `calc(0px - ${options.padding} - ${options.width})`,
+      bottom: `calc(0px - ${options.padding} - ${options.width})`,
+      left: `calc(0px - ${options.padding} - ${options.width})`,
+      right: `calc(0px - ${options.padding} - ${options.width})`,
+    },
+  };
+}
+
+const useButtonStyles = makeStyles([
+  [
+    null,
+    tokens => ({
+      alignItems: 'center',
+      display: 'inline-flex',
+      justifyContent: 'center',
+      position: 'relative',
+      verticalAlign: 'middle',
+
+      cursor: 'pointer',
+      transition: faster,
+
+      padding: `0 ${pxToRem(20)}`,
+      height: pxToRem(32),
+      minWidth: pxToRem(96),
+      maxWidth: pxToRem(280),
+
+      backgroundColor: tokens.colorScheme.default.background,
+      borderRadius: tokens.borderRadius,
+      color: tokens.colorScheme.default.foreground,
+
+      '--button-border-radius': tokens.borderWidth,
+    }),
+  ],
+
+  [{ circular: true }, { '--button-border-radius': pxToRem(4) }],
+  [{ loading: true }, { minWidth: pxToRem(118) }],
+
+  [
+    { size: 'small' },
+    {
+      padding: `0 ${pxToRem(8)}`,
+      height: pxToRem(24),
+      minWidth: pxToRem(72),
+    },
+  ],
+
+  [
+    { text: false },
```

Comment:
Why are falsey matches needed? Can't the default state represent the default look and flags define variants off the default look?

And, if there is a scenario that requires them, how would we represent "enum doesn't equal value" scenarios?

Reply:
> Is there a scenario where falsey matches are definitely needed?

To be honest, we are not 100% sure about it. It _might be useful or not_, our existing styles have a lot of negative conditions. I would also prefer to use default values and override them